marklogic-data-hub/src/main/java/com/marklogic/hub/impl/FlowManagerImpl.java

```diff
@@ -59,37 +65,27 @@
     @Autowired
     private StepDefinitionManager stepDefinitionManager;
 
+    private ArtifactService artifactService;
```

Comment:
I don't think we want this as a field of the class. Since ArtifactService is bound to a DatabaseClient, storing this as a field binds this class to a DatabaseClient, and thus to a single user. That's the main problem with all the DHF Service/Manager classes - they have HubConfig as a field, and thus they're bound to a single user. 

If the intent was for Service/Manager classes to be Spring-managed singletons, they should be stateless / threadsafe classes. That's unfortunately not the case, due to the HubConfig field. Ideally, HubConfig - or some subset of it - would be passed in as an argument to methods that need it. In the absence of that, an ArtifactService should be instantiated each time it's needed and not stored as a field.